src/main/java/org/lantern/http/InteractionServlet.java

```diff
@@ -171,11 +173,13 @@ protected void processRequest(final HttpServletRequest req,
             return;
         }
 
-        if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
-                req.getHeader("X-XSRF-TOKEN"))) {
-            log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
-            HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
-            return;
+        if (!LanternUtils.isDevMode() && !LanternUtils.isLanternPi()) {
+            if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
+                    req.getHeader("X-XSRF-TOKEN"))) {
+                log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
+                HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
+                return;
+            }
         }
```

Comment:
Okay, cool, thanks!

Cheers,
Ox

-----------------------------------------------------------------------------------------

"I love people who harness themselves, an ox to a heavy cart,
who pull like water buffalo, with massive patience,
who strain in the mud and the muck to move things forward,
who do what has to be done, again and again."

- Marge Piercy


On Fri, Jan 9, 2015 at 11:29 AM, myleshorton <notifications@github.com>
wrote:

> In src/main/java/org/lantern/http/InteractionServlet.java
> <https://github.com/getlantern/lantern/pull/2147#discussion-diff-22729425>
> :
>
> > @@ -171,11 +173,13 @@ protected void processRequest(final HttpServletRequest req,
> >              return;
> >          }
> >
> > -        if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
> > -                req.getHeader("X-XSRF-TOKEN"))) {
> > -            log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
> > -            HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
> > -            return;
> > +        if (!LanternUtils.isDevMode() && !LanternUtils.isLanternPi()) {
> > +            if (!SecurityUtils.constantTimeEquals(model.getXsrfToken(),
> > +                    req.getHeader("X-XSRF-TOKEN"))) {
> > +                log.debug("X-XSRF-TOKEN wrong: got {} expected {}", req.getHeader("X-XSRF-TOKEN"), model.getXsrfToken());
> > +                HttpUtils.sendClientError(resp, "invalid X-XSRF-TOKEN");
> > +                return;
> > +            }
> >          }
>
> Actually I guess it's really just the CSP header that causes local network
> issues -- fixed!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/lantern/pull/2147/files#r22729425>.
>